exports/tests/test_customer_export.py

```diff
@@ -0,0 +1,93 @@
+import pytest
+import xlrd
+from django.contrib.auth.models import Permission
+from django.contrib.contenttypes.models import ContentType
+from django.urls import reverse
+from rest_framework import status
+
+from berth_reservations.tests.factories import CustomerProfileFactory
+from customers.enums import InvoicingType
+from customers.models import CustomerProfile
+from customers.schema import ProfileNode
+from exports.xlsx_writer import CustomerXlsx
+from utils.relay import to_global_id
+
+
+def to_global_profile_node_ids(ids):
+    return map(lambda x: to_global_id(ProfileNode, x), ids)
+
+
+@pytest.mark.parametrize("has_permission", [True, False])
+def test_admin_credentials_are_required(user_api_client, has_permission):
+    if has_permission:
+        permission = Permission.objects.get(
+            content_type=ContentType.objects.get_for_model(CustomerProfile),
+            codename="view_customerprofile",
+        )
+        user_api_client.user.user_permissions.add(permission)
+    CustomerProfileFactory()
+    ids = CustomerProfile.objects.all().values_list("id", flat=True)
+    global_ids = to_global_profile_node_ids(ids)
+
+    response = user_api_client.post(reverse("customer_xlsx"), data={"ids": global_ids})
+
+    if has_permission:
+        assert response.status_code == status.HTTP_200_OK
+    else:
+        assert response.status_code == status.HTTP_403_FORBIDDEN
+
+
+def test_amount_of_queries(superuser_api_client, django_assert_max_num_queries):
+    CustomerProfileFactory.create_batch(2)
+    CustomerProfileFactory.create_batch(2, user=None)
+    ids = CustomerProfile.objects.all().values_list("id", flat=True)
+    global_ids = to_global_profile_node_ids(ids)
+
+    with django_assert_max_num_queries(2):
+        response = superuser_api_client.post(
+            reverse("customer_xlsx"), data={"ids": global_ids}
+        )
+
+    assert response.status_code == status.HTTP_200_OK
```

Comment:
This test is included in the test right under it.

Reply:
It is, though here it's mostly making sure that we have gotten a valid response and thus `django_assert_max_num_queries` is testing a valid situation.